gcloud-java-storage/src/test/java/com/google/gcloud/storage/ITStorageTest.java

```diff
@@ -168,13 +166,29 @@ public void testUpdateBlob() {
     assertNotNull(storage.create(blob));
     BlobInfo updatedBlob = storage.update(blob.toBuilder().contentType(CONTENT_TYPE).build());
     assertNotNull(updatedBlob);
-    assertEquals(blob.bucket(), updatedBlob.bucket());
-    assertEquals(blob.name(), updatedBlob.name());
+    assertEquals(blob.blobId(), updatedBlob.blobId());
     assertEquals(CONTENT_TYPE, updatedBlob.contentType());
     assertTrue(storage.delete(bucket, blobName));
   }
 
   @Test
+  public void testUpdateBlobReplaceMetadata() {
+    String blobName = "test-update-blob-replace-metadata";
+    BlobInfo blob = BlobInfo.builder(bucket, blobName)
+        .contentType(CONTENT_TYPE)
+        .metadata(ImmutableMap.of("k1", "a"))
+        .build();
+    assertNotNull(storage.create(blob));
```

Comment:
1) Sure
2) I'll do some digging on this. We might copy the map into an `HashMap` instead and use `Collections.unmodifiableMap` in the getter